src/cunumeric/index/zip.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/zip.h"
+#include "cunumeric/index/zip_template.inl"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <int DIM, int N>
+struct ZipImplBody<VariantKind::CPU, DIM, N> {
+  using VAL = int64_t;
+
+  template <size_t... Is>
+  void operator()(const AccessorWO<Point<N>, DIM>& out,
+                  const std::vector<AccessorRO<VAL, DIM>>& index_arrays,
+                  const Rect<DIM>& rect,
+                  const Pitches<DIM - 1>& pitches,
+                  bool dense,
+                  const int64_t key_dim,
+                  const int64_t start_index,
+                  std::index_sequence<Is...>) const
+  {
+    if (index_arrays.size() == N) {
+      const size_t volume = rect.volume();
+      if (dense) {
+        std::vector<const VAL*> indx_ptrs = {index_arrays[Is].ptr(rect)...};
+        auto outptr                       = out.ptr(rect);
+        for (size_t idx = 0; idx < volume; ++idx) {
+          outptr[idx] = Legion::Point<N>(indx_ptrs[Is][idx]...);
```

Comment:
Actually I realized this is not correct, because `Legion::Point<N>` only provides a constructor like this for up to 4 arguments.

```
cunumeric/index/zip.cc:45:25: fatal error: no matching constructor for initialization of 'Legion::Point<5>' (aka 'Point<5, long long>')
          outptr[idx] = Legion::Point<N>(indx_ptrs[Is][idx]...);
                        ^                ~~~~~~~~~~~~~~~~~~
```

Sorry, it looks like we'll need to fill these in element-by-element, as you were doing before.